pkg/internal/coordinatortest/marshaling.go

```diff
@@ -0,0 +1,255 @@
+package coordinatortest
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/keep-network/keep-core/pkg/bitcoin"
+	"github.com/keep-network/keep-core/pkg/coordinator"
+	"github.com/keep-network/keep-core/pkg/internal/hexutils"
+	"github.com/keep-network/keep-core/pkg/tbtc"
+)
+
+// UnmarshalJSON implements a custom JSON unmarshaling logic to produce a
+// proper FindDepositsToSweepTestScenario.
+func (dsts *FindDepositsToSweepTestScenario) UnmarshalJSON(data []byte) error {
+	type findDepositsToSweepTestScenario struct {
+		Title               string
+		WalletPublicKeyHash string
+		MaxNumberOfDeposits uint16
+		Wallets             []struct {
+			WalletPublicKeyHash     string
+			RegistrationBlockNumber uint64
+		}
+		Deposits []struct {
+			FundingTxHash          string
+			FundingOutputIndex     uint32
+			FundingTxConfirmations uint
+			WalletPublicKeyHash    string
+			RevealBlockNumber      uint64
+			SweptAt                int64
+		}
+		ExpectedWalletPublicKeyHash string
+		ExpectedUnsweptDeposits     []struct {
+			FundingTxHash      string
+			FundingOutputIndex uint32
+			RevealBlockNumber  uint64
+		}
+	}
+
+	var unmarshaled findDepositsToSweepTestScenario
+
+	err := json.Unmarshal(data, &unmarshaled)
+	if err != nil {
+		return err
+	}
+
+	// Unmarshal title.
+	dsts.Title = unmarshaled.Title
+
+	// Unmarshal max number of deposits.
+	if len(unmarshaled.WalletPublicKeyHash) > 0 {
+		copy(dsts.WalletPublicKeyHash[:], hexToSlice(unmarshaled.WalletPublicKeyHash))
+	}
+
+	dsts.MaxNumberOfDeposits = unmarshaled.MaxNumberOfDeposits
+
+	// Unmarshal wallets.
+	for _, wallet := range unmarshaled.Wallets {
+		w := new(Wallet)
+
+		copy(w.WalletPublicKeyHash[:], hexToSlice(wallet.WalletPublicKeyHash))
+		w.RegistrationBlockNumber = wallet.RegistrationBlockNumber
+
+		dsts.Wallets = append(dsts.Wallets, w)
+		// TODO: CHANGE TO USE i
```

Comment:
Up to you but I would be fine with the code as-is. This is not performance-critical code.